rmw_fastrtps_shared_cpp/src/rmw_wait.cpp

```diff
@@ -201,6 +201,17 @@ __rmw_wait(
         if (guard_condition.get_trigger_value()) {
           active = true;
           guard_condition.set_trigger_value(false);
+        } else {
+          switch (event->event_type) {
+            case RMW_EVENT_SUBSCRIPTION_MATCHED:
```

Comment:
It's better to add a comment to explain why we add these codes.

Reply:
Yes. I should add explanation for the change.